app/subscribers/super_good/solidus_taxjar/spree/reporting_subscriber.rb

```diff
@@ -7,12 +7,32 @@ module ReportingSubscriber
         if ::Spree::Event.method_defined?(:register)
           ::Spree::Event.register("shipment_shipped")
         end
+
         event_action :report_transaction, event_name: :shipment_shipped
+        event_action :replace_transaction, event_name: :order_recalculated
 
         def report_transaction(event)
           return unless SuperGood::SolidusTaxjar.configuration.preferred_reporting_enabled
+
           SuperGood::SolidusTaxjar::ReportTransactionJob.perform_later(event.payload[:shipment].order)
         end
+
+        def replace_transaction(event)
+          order = event.payload[:order]
+
+          return unless SuperGood::SolidusTaxjar.configuration.preferred_reporting_enabled
+
+          if order.complete? && order.paid? && amount_changed?(order)
```

Comment:
Is it possible for `replace_transaction` to be called before the order is shipped?

Reply:
I don't think that's a case we currently support. Nevertheless, in a new commit I've added a new call here that returns early if a transaction is not "replaceable". (Or: there are no previous order transactions present to replace.)